core/domain/exp_domain.py

```diff
@@ -655,6 +659,76 @@ def validate(self):
         self.outcome.validate()
 
 
+class Hint(object):
+    """Value object representing a hint."""
+    def __init__(self, hint_text, views):
+        self.hint_text = hint_text
+        self.views = views
```

Comment:
"views" should not be part of the definition of the exploration. The Exploration object should only get updated when the creator updates an exploration, and not due to any user actions.